src/content/app/genome-browser/components/zmenu/Zmenu.tsx

```diff
@@ -51,15 +59,68 @@ const Zmenu = (props: ZmenuProps) => {
 
   const destroyZmenu = () => {
     dispatch(changeHighlightedTrackId(''));
-    setZmenus && setZmenus(pickBy(zmenus, (value, key) => key !== props.id));
+    setZmenus &&
+      setZmenus(pickBy(zmenus, (value, key) => key !== props.zmenuId));
   };
 
   const direction = chooseDirection(props);
   const toolboxPosition =
     direction === Direction.LEFT ? ToolboxPosition.LEFT : ToolboxPosition.RIGHT;
 
+  const { content, variety } = props.payload;
+
+  const transcripts = content.filter(
+    (f) => f.metadata.type === ZmenuFeatureType.TRANSCRIPT
+  ) as ZmenuContentTranscript[];
```

Comment:
You are doing the same on line 89, using the more appropriate `find` method

Reply:
Remove this